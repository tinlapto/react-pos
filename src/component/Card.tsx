import { Button } from "antd";
import styled from "styled-components";

const StyledCard = styled.div`
  border: 1px solid #ddd;
  border-radius: 5px;
  flex: 0 1 20%;
  min-width: 200px;
  height: 300px;
  margin: 5px;
  display: flex;
  flex-direction: column;
  padding: 5px;
  cursor: pointer;
`;

type PhotoArgs = {
  url: string;
};
const Photo = styled.div<PhotoArgs>`
  flex: 1;
  overflow: hidden;
  border-radius: 5px;
  text-align: center;
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  background-image: url(${(prop) => prop.url});
`;

const StyledTitle = styled.div`
  text-align: center;
  padding: 10px 0;
  white-space: nowrap;
  overflow: hidden;
`;

const Card = ({
  title,
  photo,
  onClick,
}: {
  title: string;
  photo: string;
  onClick: Function;
}) => (
  <StyledCard onClick={() => onClick()}>
    <Photo url={photo} />
    <StyledTitle>{title}</StyledTitle>
    <Button style={{ width: "100%" }}>Choose</Button>
  </StyledCard>
);

export default Card;
