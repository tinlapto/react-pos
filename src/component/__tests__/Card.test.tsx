import { render, screen, fireEvent } from "@testing-library/react";
import Card from "../Card";

describe("Card", () => {
  test("render", () => {
    const onClick = jest.fn();
    const { asFragment } = render(
      <Card title={"test card"} photo={"test_url"} onClick={onClick} />
    );
    expect(asFragment()).toMatchSnapshot();
    fireEvent.click(screen.getByText("Choose"));
    expect(onClick).toBeCalled();
  });
});
