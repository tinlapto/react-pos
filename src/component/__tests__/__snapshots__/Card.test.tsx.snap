// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card render 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL egAXI"
  >
    <div
      class="sc-gsDKAQ kZkIoS"
    />
    <div
      class="sc-dkPtRN lmuKJx"
    >
      test card
    </div>
    <button
      class="ant-btn ant-btn-default"
      style="width: 100%;"
      type="button"
    >
      <span>
        Choose
      </span>
    </button>
  </div>
</DocumentFragment>
`;
